back/engines/commercial/multi_tenancy/db/seeds/community_monitor.rb

```diff
@@ -5,23 +5,34 @@
 module MultiTenancy
   module Seeds
     class CommunityMonitor < Base
+      def initialize(runner:, num_quarters: 2, ai_responses: false, locale: nil)
+        @num_quarters = num_quarters
+        @ai_responses = ai_responses
+        @locale = locale ||
+                  Locale.monolingual&.to_s ||
+                  AppConfiguration.instance.settings('core', 'locales').first ||
+                  I18n.default_locale
+        super(runner: runner)
+      end
+
       def run
-        @project = CommunityMonitorService.new.create_and_set_project
+        service = CommunityMonitorService.new
+        @project = service.project || service.create_and_set_project
         @phase = @project.phases.first
-        @question_keys = @phase.custom_form.custom_fields.reject(&:page?).pluck(:key)
-
-        # Responses for this quarter
-        runner.num_ideas.times { create_survey_response(Time.zone.now) }
-
-        # Responses for previous quarter
-        runner.num_ideas.times { create_survey_response(3.months.ago) }
+        @fields = @phase.custom_form.custom_fields.reject(&:page?)
+        @num_quarters.times do |num_quarters_ago| # NOTE: 0 = current quarter
+          runner.num_ideas.times do |index|
+            create_survey_response((3 * num_quarters_ago).months.ago, index)
+          end
+        end
       end
 
       private
 
-      def create_survey_response(created_at)
-        answers = @question_keys.index_with { |_k| rand(1..5) }
-        follow_ups = @question_keys.each_with_object({}) { |k, accu| accu["#{k}_follow_up"] = Faker::Movie.quote }
+      def create_survey_response(created_at, index)
+        question_keys = @fields.pluck(:key)
+        answers = question_keys.index_with { |_k| rand(1..5) }
+        follow_ups = question_keys.each_with_object({}) { |key, accu| accu["#{key}_follow_up"] = random_follow_up(key, index) }
```

Comment:
How do the `_follow_up` keys work? I'm wondering if we should consider adding some abstraction around this.

Reply:
`_follow_up` keys work the same way as `_other` - ie it's a text field attached to the parent field by appending `_follow_up` onto the parent field key. Not sure it's worth an abstraction here